jung-algorithms/src/main/java/edu/uci/ics/jung/layout/spatial/Node.java

```diff
@@ -77,14 +77,20 @@ public void insert(ForceObject<T> element) {
       return;
     }
     if (isLeaf()) {
-      // there already is a forceObject, so split
-      split();
-      // put the current resident and the new one into the correct quardrants
-      insertForceObject(this.forceObject);
-      insertForceObject(element);
-      // update the centerOfMass, Mass, and Force on this node
-      this.forceObject = this.forceObject.add(element);
-
+      if (this.forceObject.p.equals(element.p)) {
+        // compare points for special case where the 2 elements are at the same location
+        // this would cause an infinite attempt to split and re-insert
+        // just add the new mass
+        this.forceObject = this.forceObject.add(element);
```

Comment:
I don't know enough about the BH algorithm to know whether this is the right answer, or whether it would be more appropriate to perturb the new element's position by a small random value and attempt re-insertion.  Are there any potential pitfalls of identifying two ForceObjects as a single object?

Reply:
OK, thanks for clarifying.